src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -750,6 +751,39 @@ private Object decodeBasicOther(
                 position.set(Position.KEY_ARCHIVE, buf.readUnsignedByte() > 0);
             }
 
+        } else if (type == MSG_ALARM) {
+
+            DateBuilder dateBuilder = new DateBuilder(deviceSession.getTimeZone())
+                    .setDate(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte())
+                    .setTime(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte());
+
+            getLastLocation(position, dateBuilder.getDate());
+
+            short alarmType = buf.readUnsignedByte();
+
+            switch (alarmType) {
+                case 0x80:
+                    position.set(Position.KEY_ALARM, Position.ALARM_VIBRATION);
+                    break;
+                case 0x87:
+                    position.set(Position.KEY_ALARM, Position.ALARM_OVERSPEED);
+                    break;
+                case 0x90:
+                    position.set(Position.KEY_ALARM, Position.ALARM_ACCELERATION);
+                    break;
+                case 0x91:
+                    position.set(Position.KEY_ALARM, Position.ALARM_BRAKING);
+                    break;
+                case 0x92:
+                    position.set(Position.KEY_ALARM, Position.ALARM_CORNERING);
+                    break;
+                default:
+                    position.set(Position.KEY_ALARM, Position.ALARM_GENERAL);
+                    break;
+            }
+
+            position.set("alarmValue", buf.readShort());
```

Comment:
What's alarm value?

Reply:
If you look in the specification linked above, on pages 23-24, section 5.4.1. they say that the message contains

* Alarm Type (1 byte)
* Alarm Value (N bytes) 

In sections 5.4.1.6 and 5.4.1.7 they say it can contain speed value for over-speed alert. 

I tested all the alerts while driving the car and the "Alarm Value" is always 2 byte number - the mentioned speed for overspeed alert or a number for the rest. I assume those values correspond with alert sensitivity, because the appliance can be set to 3 different levels of sensitivity for acceleration/deceleration/turn (see commands settings on page 43+).

The specification does not say much more about it and is a bit chaotic, probably machine translated from Chinese.